tower/src/util/future_service.rs

```diff
@@ -0,0 +1,163 @@
+use std::fmt;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower_service::Service;
+
+/// Returns a new `FutureService` for the given future.
+///
+/// A `FutureService` allows you to treat a future that resolves to a service as a service. This
+/// can be useful for services that are created asynchronously.
+///
+/// # Example
+/// ```
+/// use tower::{service_fn, Service, ServiceExt};
+/// use tower::util::future_service;
+/// use std::convert::Infallible;
+///
+/// # fn main() {
+/// # async {
+/// let mut future_of_a_service = future_service(Box::pin(async {
+///     let svc = service_fn(|_req: ()| async { Ok::<_, Infallible>("ok") });
```

Comment:
take it or leave it: it _might_ be worth showing this actually await something? otherwise, the example code shows an unnecessary future (this could just synchronously return the service). we could even just have an `async fn new_service` or something whose body is hidden. just a thought.